### PR TITLE
Only show `no-undef` errors for templates in gts files

### PR DESCRIPTION
**Pull request by patricklx**
this filters out any linting message that is within the template, except for the no-undef rule.
there are also errors like `Expected blank line between class members`, which wrap the whole template block, which will be included

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
Can tests be added? Thanks!

Also, is no-undef really the only lint we care about? 😅

**Comment by patricklx** [comment]
Yes, according to this comment: https://github.com/ember-cli/eslint-plugin-ember/blob/208e6b6e8a017f84671891c42953913a5eb05e3f/lib/preprocessors/glimmer.js#L88

**Comment by NullVoxPopuli** [comment]
hmm I wrote that. :upside_down_face: 

Sounds good then -- ping when you push up some tests or if you have any questions -- will be good to get this merged!

**Comment by NullVoxPopuli** [comment]
Nice, lgtm then. @bmish , can you approve the workflow run?

File: lib/preprocessors/glimmer.js
Comment (bmish) [inline lib/preprocessors/glimmer.js:165]: Can you comment to explain why `no-undef` is the only rule included here?

Comment (patricklx) [reply lib/preprocessors/glimmer.js:165]: i think @NullVoxPopuli should answer :)

File: lib/preprocessors/glimmer.js
Comment (patricklx) [reply lib/preprocessors/glimmer.js:11]: maybe the naming is not ideal. how about `TEMPLATE_RANGE_CACHE`. it saves the template range so we can know which eslint messages are inside the template ranges in the postprocessing step

Comment (patricklx) [reply lib/preprocessors/glimmer.js]: i moved the whole thing to postprocessing, as the cache is not really needed

File: lib/preprocessors/glimmer.js
Comment (hmajoros) [reply lib/preprocessors/glimmer.js:164]: Is the logic here "ember-template-lint should give us errors from inside a template context"?

are we certain there will not be other eslint rules written in the future that may require us to update this logic?

Comment (NullVoxPopuli) [reply lib/preprocessors/glimmer.js:164]: @hmajoros exactly -- the only thing template-lint can't know about is what is in scope -- so the template-lint equiv of no-undef is _disabled_ for gjs/gts

File: lib/preprocessors/glimmer.js
Comment (bmish) [inline lib/preprocessors/glimmer.js:1]: Is this new code covered by tests? Running `yarn test --coverage`, I see a lot of uncovered lines.

File: lib/preprocessors/glimmer.js
Comment (patricklx) [reply lib/preprocessors/glimmer.js:164]: There is a comment here already: https://github.com/ember-cli/eslint-plugin-ember/blob/208e6b6e8a017f84671891c42953913a5eb05e3f/lib/preprocessors/glimmer.js#L88

Comment (patricklx) [reply lib/preprocessors/glimmer.js:164]: Should i move it down to this code?

File: lib/preprocessors/glimmer.js
Comment (patricklx) [reply lib/preprocessors/glimmer.js:1]: You are right, Not every line, no. I can enable one more rule

File: lib/preprocessors/glimmer.js
Comment (bmish) [reply lib/preprocessors/glimmer.js:164]: I think two improvements would help:

1. Update the comment to explain why we need to "allow-list the rules that check for undefined identifiers".
2. Move the rule name into an array with a descriptive name:

```js
const RULES_THAT_CHECK_FOR_UNDEFINED_IDENTIFIERS = ['no-undef'];
```

This name is self-documenting and an array would be reusable and easily expandable.

**Comment by patricklx** [comment]
mmm, i will rework this to use the DocumentLines utils from my other PR, which makes the check more easier.
just noticed the check is not completely correct, would be easier to test with offsets.

File: lib/preprocessors/glimmer.js
Comment (bmish) [inline lib/preprocessors/glimmer.js:32]: I still would like to see an explanation for why this is the only rule we care about.

Comment (patricklx) [reply lib/preprocessors/glimmer.js:32]: there is here: https://github.com/ember-cli/eslint-plugin-ember/blob/208e6b6e8a017f84671891c42953913a5eb05e3f/lib/preprocessors/glimmer.js#L84.
. Should i add an explanation here too? Like
"Because otherwise eslint would show errors unrelated to the original code and apply autofixes like spacing, comna etc into the template"

**Comment by patricklx** [comment]
@bmish fixed, tests pass, full coverage

File: lib/preprocessors/glimmer.js
Comment (bmish) [reply lib/preprocessors/glimmer.js:32]: Any where we could clarify comments is appreciated. I will leave it to your and @NullVoxPopuli's judgment.

File: lib/utils/document.js
Comment (patricklx) [reply lib/utils/document.js:5]: It comes from here: https://github.com/typed-ember/glint/blob/main/packages/core/src/language-server/util/position.ts

And will also be used for 
https://github.com/ember-cli/eslint-plugin-ember/pull/1853

File: lib/preprocessors/glimmer.js
Comment (NullVoxPopuli) [reply lib/preprocessors/glimmer.js]: things look good from over here

**Comment by patricklx** [comment]
Coverage was not enough. I think because of the document utility class.
Can we disable the coverage for the document utility? It has a function which i do not use now. But i use it in the other pr.
I can try to enable it there again

**Review by bmish — APPROVED** [review]
Thanks, merging to unblock #1853.

**Pull request merged**
